CMakeLists.txt

```diff
@@ -120,8 +124,20 @@ endif()
 #
 # Looking for external libraries
 #
-
-setup_conan()
+if(USE_CONAN)
+	setup_conan()
+else("Searching for required libraries using CMake")
+	find_package(muparserx REQUIRED)
+	find_package(nlohmann_json REQUIRED)
+	find_package(spdlog REQUIRED)
+
+	# Multiprocessing/Thrust libraries
+	find_package(llvm-openmp)
+	find_package(tbb)
+
+	# for tests only
+	find_package(catch2)
```

Comment:
Some of these dependencies are only needed for certain build paths. For example, `muparserx` is only needed for the python addon, but not for the standalone version. `llvm-openmp` is only needed in Macos , etc. Ideally we should define which dependencies we really need depending on the build and then find them with conan or without it.